opentelemetry-collector.yaml

```diff
@@ -21,13 +21,13 @@ pipeline:
       # https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/distributions/otelcol/manifest.yaml
       # The manifest is updated as part of the release process and so should match the
       # lastest version released from open-telemetry/opentelemetry-collector
-      curl -o builder-config.yaml https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector-releases/main/distributions/otelcol/manifest.yaml
+      curl -o builder-config.yaml https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector-releases/v${{package.version}}/distributions/otelcol/manifest.yaml
```

Comment:
@rawlingsj This version was not released yet in `/opentelemetry-collector-releases` that is why I mentioned it requires other dependencies to release v0.105.0 too.

Reply:
but we should use that url for our current version v0.104.0 in main.

Reply:
v0.105.0 is now ready 🎉